playground/infrastructure/helper.py

```diff
@@ -112,24 +113,27 @@ def find_examples(work_dir: str, supported_categories: List[str],
     """
     has_error = False
     examples = []
-    for root, _, files in os.walk(work_dir):
-        for filename in files:
-            filepath = os.path.join(root, filename)
-            error_during_check_file = _check_file(
-                examples=examples,
-                filename=filename,
-                filepath=filepath,
-                supported_categories=supported_categories,
-                sdk=sdk)
-            has_error = has_error or error_during_check_file
+    for subdir in subdirs:
```

Comment:
what if someone puts some dir and it's subdir to list? Are we going to fail with duplicate examples?

Reply:
yep, we are

Reply:
This failure will only mean that `SUBDIRS` variable is set incorrectly in GH workflow, and the fix is needed

Reply:
Though, I'd rather add this check as it raises concerns

Reply:
@damccorm 
After 5 minutes I realized that it's not trivial to make this check
Naive version I've reverted blocks valid cases like: `["sub", "subsub"]
Of course, we could split it with `os.path.split` but it won't normalize paths like `sub1/../sub2`
More elaborated version could use `os.path.normpath` helper to normalize paths, but still won't recognize cases like
`[".", "sub"]`

I feel like that the profit won't justify the effort here

Reply:
Could we just make a map of filepaths to subdirs and check that as we walk the filesystem? If a map key already exists we can throw an error with the 2 offending subdirs listed.

Reply:
Let me explain the solution.
If subdirs are sorted alphabetically:
```
01sub/sub
02sub
02sub/subsub
```
 it allows to check only pairs of adjacent items, and we don't need to store items

Reply:
Technically this probably doesn't handle situations like `abc/def` is a parent of `./abc/def/ghi` (and other relative path cases). Otherwise, this seems reasonable; maybe we can just convert everything to PurePaths before sorting and then sort by parents?